### PR TITLE
Enable GitHub release

### PR DESCRIPTION
**Pull request by ljubon**
The goal of this PR is to add a release job with a target for GitHub
The Nuget Release job is already present, the same package/artifacts are used to publish it to the GitHub release

https://github.com/G-Research/oss-portfolio-maturity/issues/332

### CODE REVIEW

**Comment by Smaug123** [comment]
Does anyone actually get any value from GitHub releases for NuGet packages? I claim the answer is "no"; is there any evidence that they do?

**Comment by nojaf** [comment]
In other projects I have seen that some people subscribe to releases:
image

If you cannot use something like dependabot for example...

**Comment by pavlovic-ivan** [comment]
Yes, and i've also seen other projects like winsw that are strictly C# (meaning not other languages used in the project), that publish to Github releases. But like i've said, if in general we don't want this, that's ok

File: .github/workflows/dotnet.yaml [hunk 1]
```diff
@@ -141,3 +141,20 @@ jobs:
...
       - name: Publish to NuGet
         run: dotnet nuget push "ApiSurface.*.nupkg" --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
+
+  github-release:
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.repository.fork && github.ref == 'refs/heads/main' }}
+    needs: [all-required-checks-complete]
+    environment: release
+    permissions:
+      contents: write
```
Comment (Smaug123) [inline .github/workflows/dotnet.yaml:151]: Seems a bit sad that GitHub's perms model isn't more finely grained here. This gives permission to update the commit hash of the `main` branch, for example.

**Review by Smaug123 — APPROVED** [review]
I guess it does no harm

File: .github/workflows/dotnet.yaml (hunk 1 above)
Comment (EnricoMi) [reply .github/workflows/dotnet.yaml:151]: This job executes external code that you do not have control of: `actions/download-artifact@v4` and `ncipollo/release-action@v1`. Attacks on that code will be able to make arbitrary use of the `write` permission.

It is good practice to pin actions to a specific commit (not a tag like `v1` that can change over time) plus a thorough review of that commit's code.

**Review by nojaf — APPROVED** [review]
I'm approving because Patrick did.
This looks good, but I lack a strong opinion on this.

File: .github/workflows/dotnet.yaml (hunk 1 above)
Comment (EnricoMi) [reply .github/workflows/dotnet.yaml:151]: Dependabot supports those commit versions as well, it keeps suggesting latest versions (here commits): https://github.com/EnricoMi/publish-unit-test-result-action/pull/473

Comment (pavlovic-ivan) [reply .github/workflows/dotnet.yaml:151]: We haven't reached a point where we should distrust the tags yet. If we did, a problem would be a lot bigger than what we are discussing here, and there would hundreds of PRs opened in this org alone to mitigate the issue. I do agree with the practice, but not so far to block PR from merging (i am aware it's approved). I will add though that when it comes to github actions (official and 3rd party), that we plan to have proxy actions in place of those, where we would do exactly what is mentioned (pin to a commit, and update to the next if review of the changes allows it). That will generate a lot of PRs over time as expected. I hope this gives some level of serenity

Comment (Smaug123) [reply .github/workflows/dotnet.yaml:151]: Having just checked what that action does, a single line of curl would be simpler and easier to review :P

Comment (pavlovic-ivan) [reply .github/workflows/dotnet.yaml:151]: True that 😅

**Comment by Smaug123** [comment]
Please revert this; it's broken. Life's too short to work out why, since nobody actually requested this.

**Pull request merged**